### DISCUSSION TITLE
Move autogenerated files to .yarn folder

### DISCUSSION THREAD

**Discussion by thefrana** [yarnpkg/berry#3689] [Q&A] [answered]
Hi,
I really like the idea of Yarn. It's great. One thing that is bothering me is that it creates a lot of files.

```
- .yarn/
- .yarnrc.yml
- .pnp.cjs
- .pnp.loader.mjs
- yarn.lock
```

Is it possible to move last three (auto-generated) files to `.yarn/`? I would keep only `yarnrc.yml` in the root folder and specify all locations of the moved files in the config. Or can I move even the config file to `.yarn/`?

Thanks

**Comment by arcanis** [comment, marked as answer]
`.yarnrc.yml` can't move, since it follows a typical rc directory hierarchy structure.

`.pnp.cjs` can't move, since various tools expect it to be at the root in order to feature-detect PnP.

`yarn.lock` can't move, for the same reason.

`.pnp.loader.mjs` could potentially move, since it isn't used a lot yet.

**Reply by thefrana** [reply]
So basically nothing can be moved. I think the root project folder is bloated that way, but I understand it. Thank you.

**Reply by mikestopcontinues** [reply]
@arcanis I still think there's a case for a breaking change with the next version. The presence of a `.yarn` dir indicates this is a package. Requiring five items in a directory view is a lot.
